### PR TITLE
Add config overrides for ironic

### PR DESCRIPTION
**Pull request by cjloader**
This adds the necessary variables for ironic to communicate with
glance for image upload/retrieval.

Issue: RI-313

Issue: [RI-313](https://rpc-openstack.atlassian.net/browse/RI-313)

### CODE REVIEW

File: etc/openstack_deploy/group_vars/all/osa.yml [hunk 1]
```diff
@@ -96,3 +96,14 @@ haproxy_extra_services:
 
 # Define the distro version globally
 repo_build_os_distro_version: "{{ (ansible_distribution | lower) | replace(' ', '_') }}-{{ ansible_distribution_version.split('.')[:2] | join('.') }}-{{ ansible_architecture | lower }}"
+
+# remove once https://review.openstack.org/#/c/573822/ merges
+ironic_ironic_conf_overrides:
+  glance:
+    auth_url: "http://{{ internal_lb_vip_address }}:35357/v3"
+    auth_type: "password"
```
Comment (prometheanfire) [inline etc/openstack_deploy/group_vars/all/osa.yml]: like the upstream review, most of these things should be variables, not hard coded. (domain, project, user, auth_type)

**Comment by prometheanfire** [comment]
also, are these being tracked for removal as well (for when the fix is upstreamed)?

File: etc/openstack_deploy/group_vars/all/osa.yml [hunk 2]
```diff
@@ -96,3 +96,15 @@ haproxy_extra_services:
 
 # Define the distro version globally
 repo_build_os_distro_version: "{{ (ansible_distribution | lower) | replace(' ', '_') }}-{{ ansible_distribution_version.split('.')[:2] | join('.') }}-{{ ansible_architecture | lower }}"
+
+# remove once https://review.openstack.org/#/c/573822/ merges
+ironic_ironic_conf_overrides:
+  glance:
+    auth_url: http://{{ internal_lb_vip_address }}:35357/v3
+    auth_type: " {{ glance_keystone_auth_plugin }}"
```
Comment (prometheanfire) [inline etc/openstack_deploy/group_vars/all/osa.yml]: extra space before `{{`

File: etc/openstack_deploy/group_vars/all/osa.yml [hunk 3]
```diff
@@ -96,3 +96,15 @@ haproxy_extra_services:
 
 # Define the distro version globally
 repo_build_os_distro_version: "{{ (ansible_distribution | lower) | replace(' ', '_') }}-{{ ansible_distribution_version.split('.')[:2] | join('.') }}-{{ ansible_architecture | lower }}"
+
+# remove once https://review.openstack.org/#/c/573822/ merges
```
Comment (prometheanfire) [inline etc/openstack_deploy/group_vars/all/osa.yml]: do we have a text blob like `TODO` for items needing cleanup?  is there a jira issue for the removal?

**Comment by cjloader** [comment]
Works now.

**Pull request merged**